client-service-worker/gulpfile.js

```diff
@@ -0,0 +1,66 @@
+const gulp = require('gulp');
+const gap = require('gulp-append-prepend');
+const del = require('del');
+const merge2 = require('merge2');
+
+const {NODE_ENV} = process.env;
+
+const isProd = NODE_ENV === 'production';
+const isTest = NODE_ENV === 'test';
+
+const JSDELIVR_CDN_DOMAIN = process.env.JSDELIVR_CDN_DOMAIN ? process.env.JSDELIVR_CDN_DOMAIN.trim() : 'cdn.jsdelivr.net';
+
+const hostnames = {
+  jsdelivrCdnDomain: JSDELIVR_CDN_DOMAIN
+};
+
+const HOST_NAMES = `;var HOST_NAMES = ${JSON.stringify(hostnames)};`;
+
+
+async function buildApp() {
+  return merge2(
+    gulp.src('src/**/*.js', { sourcemaps: !isProd, since: gulp.lastRun(buildApp)})
```

Comment:
Suggest to skip sourcemaps in gulp.src and gulp.dest, I think they are not needed for this simple prepend.